components/ui/VersionFilterControl.vue

```diff
@@ -131,8 +154,37 @@ export default {
               projectVersion.loaders.includes(loader)
             ))
       )
+      this.updateQuery()
       this.$emit('updateVersions', temp)
     },
+    validateFilters() {
+      this.selectedLoaders = this.selectedLoaders.filter((loader) =>
+        this.getValidLoaders().includes(loader)
+      )
+      this.selectedGameVersions = this.selectedGameVersions.filter((version) =>
+        this.getValidVersions().some(
+          (validVersion) => validVersion.version === version
+        )
+      )
+    },
+    updateQuery() {
+      this.$router
+        .replace({
+          query: {
+            ...this.$route.query,
+            loaders: this.getAsQuery(this.selectedLoaders),
+            gameVersions: this.getAsQuery(this.selectedGameVersions),
+            showSnapshots: this.showSnapshots ? true : undefined,
+          },
+        })
+        .catch(() => {})
+    },
+    getAsQuery(elements) {
+      if (elements.length === 0) {
+        return undefined
+      }
+      return elements.join(',')
+    },
```

Comment:
This shouldn't be a separate function if it is only called once. The parent function should be `async` and await this promise instead of having an empty catch block

Reply:
ignore that this comment contains `getAsQuery`, accidentally selected too much

Reply:
> This shouldn't be a separate function if it is only called once.

It is called by `updateVersionFilters` and from the include snapshots checkbox.

Reply:
> The parent function should be async and await this promise instead of having an empty catch block

Since the `updateQuery` is now also called in the event of the checkbox instead of the watch, you can't await it there.

Reply:
> > The parent function should be async and await this promise instead of having an empty catch block
> 
> Since the `updateQuery` is now also called in the event of the checkbox instead of the watch, you can't await it there and creating a separate method for that doesn't seem to be reasonable.

This isn't how Vue does stuff, you don't have to await in an event for it to still be run async. You should make it async and await it- that's the codestyle this repo uses